pkg/controller/jaeger/account.go

```diff
@@ -0,0 +1,59 @@
+package jaeger
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/inventory"
+)
+
+func (r *ReconcileJaeger) applyAccounts(jaeger v1alpha1.Jaeger, desired []v1.ServiceAccount) error {
+	opts := client.MatchingLabels(map[string]string{
+		"app.kubernetes.io/instance":   jaeger.Name,
+		"app.kubernetes.io/managed-by": "jaeger-operator",
+	})
+	list := &v1.ServiceAccountList{}
+	if err := r.client.List(context.Background(), opts, list); err != nil {
+		return err
+	}
+
+	inv := inventory.ForAccounts(list.Items, desired)
+	for _, d := range inv.Create {
+		log.WithFields(log.Fields{
```

Comment:
This again seems a repetitive and makes 30% of the file. We could use this https://github.com/sirupsen/logrus#default-fields and pass logger around. Or at least have a util function that returns logger with default fields.

Reply:
This is how it would look like:

```golang
	logFields := log.WithFields(log.Fields{
		"namespace": jaeger.Namespace,
		"instance":  jaeger.Name,
	})
```

And then, within each loop iteration:

```golang
		logFields.WithField("account", d.Name).Debug("creating service account")
```

It does look better, just not sure how much better :) If you prefer it this way, I have no problems in changing it.